massa-bootstrap/src/tests/scenarios.rs

```diff
@@ -177,24 +183,22 @@ async fn test_bootstrap_server() {
     });
 
     // accept connection attempt from remote
-    let (remote_rw, conn_addr, resp) = tokio::time::timeout(
-        std::time::Duration::from_millis(1000),
-        remote_interface.wait_connection_attempt_from_controller(),
-    )
-    .await
-    .expect("timeout waiting for connection attempt from remote")
-    .expect("error receiving connection attempt from remote");
-    let expect_conn_addr = bootstrap_config.bootstrap_list[0].0;
-    assert_eq!(
-        conn_addr, expect_conn_addr,
-        "client connected to wrong bootstrap ip"
-    );
-    resp.send(true)
-        .expect("could not send connection accept to remote");
+    let remote_bridge = std::thread::spawn(move || {
+        let (remote_rw, conn_addr, waker) = remote_interface
+            .wait_connection_attempt_from_controller()
+            .expect("timeout waiting for connection attempt from remote");
+        let expect_conn_addr = bootstrap_config.bootstrap_list[0].0;
+        assert_eq!(
+            conn_addr, expect_conn_addr,
+            "client connected to wrong bootstrap ip"
+        );
+        waker.store(true, Ordering::Relaxed);
+        remote_rw
+    });
 
     // connect to bootstrap
     let remote_addr = std::net::SocketAddr::from_str("82.245.72.98:10000").unwrap(); // not checked
```

Comment:
what is this address ?

Reply:
it's tcp stream of the mocked client. When calling `get_state` the requests _to_ the bootstrap are sent _from_ this stream (At least, that's my best understanding. This system is pretty tough to reason about. #3730 removes it alltogether, pretty much.